book/website/reproducible-research/licensing.md

```diff
@@ -10,102 +13,301 @@ No previous knowledge is needed, this chapter explains how important it is to un
+Consequently, these licence by the classical definitions of free and open-source software from the FSF and OSI would not be considered free or open-source licences, they do however generally resemble them in the other three criteria of the definition.
+Their merits versus conventional open-source licenses have been the subject of some debate, and their adoption has thus far been relatively limited.
+
+Even an attribution requirement (the BY in CC-BY) can in some cases be considered a usage restriction.
+For example the Debian project found the [Common Public Attribution License (CPAL)](https://en.wikipedia.org/wiki/Common_Public_Attribution_License) to be incompatible their free-software guidelines for this reason whilst it is approved by the Open Source Initiative.
+In the case of academic works attribution requirements can serve to re-enforce the citation convention with the force of copyright law.
+
+(rr-licensing-choosing)=
+### Where to find open licenses for different types of work
+
+- Code
+    - The [Open Source Initiaitive (OSI)](https://opensource.org/licenses/) maintains a list of [approved Licences](https://opensource.org/licenses/) open-source licences
+    - [Free Software Foundation](https://www.fsf.org/) maintains a [list of GPL-Compatible Free Software Licenses](https://www.gnu.org/licenses/license-list.html#SoftwareLicenses)
+        - [GNU/FSF recomendations](https://www.gnu.org/licenses/license-recommendations.html)
+    - [choosealicense.com](https://choosealicense.com/) provides a tool to guide you through the license choice project.
+    - [Organisation for ethical source](https://ethicalsource.dev/) maintains a list of [ethical source licenses](https://ethicalsource.dev/licenses/)
+- Prose, Images, Audio, Video, Datasets, and similar
+    - [Creative Commons (CC)](https://creativecommons.org/)
+        - [Creative Commons License Chooser](https://creativecommons.org/choose/)
+          ![figure from https://libguides.gwu.edu/opentextbooks/creative_commons](https://s3.amazonaws.com/libapps/accounts/22603/images/CC-License-Chart.png)
+          <!-- alt text needed! -->
+- Machine Learning (ML) / artificial inteligence (AI) systems
+    - Creative commons and Software licences can be applied to different parts of ML/AI systems, CC to training data and weights, software licences to code used in training / depoyment.
+    - [Responsible AI Licenses (RAIL)](https://www.licenses.ai/)
+
+(rr-licensing-enforcement)=
+### Licencing enforcement
+
+There have been a number of successful legal cases that have been brought in defence of the terms of copyleft licenses obliging the parties abusing the terms of these licenses to appropriately release their code.
+But this can be hard to discover, as it is not immediately obvious if copyleft code has been used from looking at a black box proprietary end product.
```

Comment:
I think we are being inconsistent using "copyleft" and "copy-left".

Reply:
Yes changing all to "copyleft" for reasons stated in: [23cc7f5](https://github.com/the-turing-way/the-turing-way/pull/3142/commits/23cc7f53735340a0b32763e7025b8b54ac67ef76)